frontend/src/concepts/pipelines/content/artifacts/ArtifactUriLink.tsx

```diff
@@ -11,58 +11,99 @@ interface ArtifactUriLinkProps {
 }
 
 export const ArtifactUriLink: React.FC<ArtifactUriLinkProps> = ({ artifact }) => {
-  const isClassificationMetrics = artifact.getType() === ArtifactType.CLASSIFICATION_METRICS;
+  const isMetrics =
+    artifact.getType() === ArtifactType.CLASSIFICATION_METRICS ||
+    artifact.getType() === ArtifactType.METRICS ||
+    artifact.getType() === ArtifactType.SLICED_CLASSIFICATION_METRICS;
+
+  const isDownloadableOnly = artifact.getType() === ArtifactType.MODEL;
+  const { getStorageObjectDownloadUrl, getStorageObjectRenderUrl } = useArtifactStorage();
   const [loading, setLoading] = React.useState<boolean>(false);
 
-  const { getStorageObjectUrl } = useArtifactStorage();
+  const handleDownload = () => {
+    getStorageObjectDownloadUrl(artifact)
+      .then((url) => {
+        if (url) {
+          triggerFileDownload('', url);
+        }
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error('Failed to get the storage URL:', error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  };
 
-  const handleOnClick = async () => {
-    const url = await getStorageObjectUrl(artifact);
-    return url;
+  const handlePreview = () => {
+    getStorageObjectRenderUrl(artifact)
+      .then((url) => {
+        if (url) {
+          window.open(url, '_blank');
+        }
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error('Failed to get the storage URL:', error);
```

Comment:
do you think we should have notifications for these errors instead of silencing them

Reply:
Yes, I will change it to a notification.